internal/services/functions/build_subscription_scope_resource_id_test.go

```diff
@@ -0,0 +1,129 @@
+package functions_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/terraform-provider-azapi/internal/services/functions"
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/function"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func Test_SubscriptionResourceIdFunction(t *testing.T) {
+	testCases := map[string]struct {
+		request  function.RunRequest
+		expected function.RunResponse
+	}{
+		"subscription-scope-valid-single": {
+			request: function.RunRequest{
+				Arguments: function.NewArgumentsData([]attr.Value{
+					types.StringValue("00000000-0000-0000-0000-000000000000"),
+					types.StringValue("Microsoft.Sql/locations@2015-05-01-preview"),
```

Comment:
I think similar to above, this relies on the api version to get schema info, also considering staying uniform with other resource id functions, I think we should keep the api version

Reply:
> this relies on the api version to get schema info
Could you elaborate on this? 

>  also considering staying uniform with other resource id functions, I think we should keep the api version
I think we could remove the api-version from all the resource id functions, because it's not used.

Reply:
The api version is used to get the schema info stored in the cache (json) about this resourcetype which then used to verify whether the parentId matches with the expectation given the scope type of this resourcetype. 
Do you reckon this validation is not needed?

Reply:
Thanks for the details! Yes, I think we can skip the scope validation.